Core/include/Acts/Seeding/GbtsGeometry.hpp

```diff
@@ -35,330 +33,63 @@ class TrigInDetSiLayer {
   std::vector<float> m_maxRadius;
   std::vector<float> m_minBinCoord;
   std::vector<float> m_maxBinCoord;
 
-  float m_minEta{}, m_maxEta{};
+  float m_minEta, m_maxEta, m_etaBin;
 
  protected:
-  float m_etaBinWidth{}, m_phiBinWidth{};
+  float m_etaBinWidth;
 
-  float m_r1{}, m_z1{}, m_r2{}, m_z2{};
-  float m_nBins{};
-  float m_etaBin{};
+  float m_r1, m_z1, m_r2, m_z2;
+  int m_nBins;
 };
 
-template <typename space_point_t>
 class GbtsGeometry {
  public:
-  GbtsGeometry(const std::vector<TrigInDetSiLayer> &layers,
-               std::unique_ptr<GbtsConnector> &conn)
-
-      : m_connector(std::move(conn)) {
-    const float min_z0 = -168.0;
-    const float max_z0 = 168.0;
-
-    m_etaBinWidth = m_connector->m_etaBin;
-    for (const auto &layer : layers) {
-      const GbtsLayer<space_point_t> *pL = addNewLayer(layer, m_nEtaBins);
-      m_nEtaBins += pL->num_bins();
-    }
-
-    // calculating bin tables in the connector...
-
-    for (auto &[_, vConn] : m_connector->m_connMap) {
-      for (auto &c : vConn) {
-        unsigned int src = c->m_src;  // n2 : the new connectors
-        unsigned int dst = c->m_dst;  // n1
-
-        const GbtsLayer<space_point_t> *pL1 = getGbtsLayerByKey(dst);
-        const GbtsLayer<space_point_t> *pL2 = getGbtsLayerByKey(src);
-
-        if (pL1 == nullptr) {
-          std::cout << " skipping invalid dst layer " << dst << std::endl;
-          continue;
-        }
-        if (pL2 == nullptr) {
-          std::cout << " skipping invalid src layer " << src << std::endl;
-          continue;
-        }
-        int nSrcBins = pL2->m_bins.size();
-        int nDstBins = pL1->m_bins.size();
+  GbtsGeometry(const std::vector<TrigInDetSiLayer>& layers,
+               const std::unique_ptr<GbtsConnector>& conn);
+  ~GbtsGeometry();
 
-        c->m_binTable.resize(nSrcBins * nDstBins, 0);
-
-        for (int b1 = 0; b1 < nDstBins; b1++) {    // loop over bins in Layer 1
-          for (int b2 = 0; b2 < nSrcBins; b2++) {  // loop over bins in Layer 2
-            if (!pL1->verifyBin(pL2, b1, b2, min_z0, max_z0)) {
-              continue;
-            }
-            int address = b1 + b2 * nDstBins;
-            c->m_binTable.at(address) = 1;
-          }
-        }
-      }
-    }
-  }
-
-  GbtsGeometry() = default;
-
-  // for safety to prevent passing as copy
-  GbtsGeometry(const GbtsGeometry &) = delete;
-  GbtsGeometry &operator=(const GbtsGeometry &) = delete;
-
-  const GbtsLayer<space_point_t> *getGbtsLayerByKey(unsigned int key) const {
-    typename std::map<unsigned int, GbtsLayer<space_point_t> *>::const_iterator
-        it = m_layMap.find(key);
-    if (it == m_layMap.end()) {
-      return nullptr;
-    }
-
-    return (*it).second;
-  }
-
-  const GbtsLayer<space_point_t> *getGbtsLayerByIndex(int idx) const {
-    return m_layArray.at(idx).get();
-  }
+  const GbtsLayer* getGbtsLayerByKey(unsigned int key) const;
+  const GbtsLayer* getGbtsLayerByIndex(int idx) const;
```

Comment:
- can this rather return a reference or is `nullptr` meaningful? i.e. "entry not found"
- rather use `std::uint32_t` and `std::int32_t`